qiskit/aqua/operators/gradients/hessian.py

```diff
@@ -0,0 +1,258 @@
+            # Compute the Hessian entries corresponding to these pairs of parameters.
+            elif all(isinstance(param, tuple) for param in params):
+                return ListOp([self.convert(operator, param_pair) for param_pair in params])
+
+        expec_op = PauliExpectation(group_paulis=False).convert(operator).reduce()
+        cleaned_op = self._factor_coeffs_out_of_composed_op(expec_op)
+        return self.get_hessian(cleaned_op, params)
+
+    # pylint: disable=too-many-return-statements
+    def get_hessian(self,
+                    operator: OperatorBase,
+                    params: Optional[Union[Tuple[ParameterExpression, ParameterExpression],
+                                           List[Tuple[ParameterExpression, ParameterExpression]]]]
+                    = None
+                    ) -> OperatorBase:
+        """Get the Hessian for the given operator w.r.t. the given parameters
+
+        Args:
+            operator: Operator w.r.t. which we take the Hessian.
+            params: Parameters w.r.t. which we compute the Hessian.
+
+        Returns:
+            Operator which represents the gradient w.r.t. the given params.
+
+        Raises:
+            ValueError: If ``params`` contains a parameter not present in ``operator``.
+            AquaError: If the coefficient of the operator could not be reduced to 1.
+                        AquaError: If the differentiation of a combo_fn requires JAX but the package
+                        is not installed.
+            TypeError: If the operator does not include a StateFn given by a quantum circuit
+            Exception: Unintended code is reached
+        """
+
+        def is_coeff_c(coeff, c):
+            if isinstance(coeff, ParameterExpression):
+                expr = coeff._symbol_expr
+                return expr == c
+            return coeff == c
+
+        if isinstance(params, (ParameterVector, list)):
+            # Case: a list of parameters were given, compute the Hessian for all param pairs
+            if all(isinstance(param, ParameterExpression) for param in params):
+                return ListOp(
+                    [ListOp([self.get_hessian(operator, (p0, p1)) for p1 in params])
+                     for p0 in params])
+            # Case: a list was given containing tuples of parameter pairs.
+            # Compute the Hessian entries corresponding to these pairs of parameters.
+            elif all(isinstance(param, tuple) for param in params):
+                return ListOp(
+                    [self.get_hessian(operator, param_pair) for param_pair in params])
+
+        # If a gradient is requested w.r.t a single parameter, then call the
+        # Gradient().get_gradient method.
+        if isinstance(params, ParameterExpression):
+            return Gradient(grad_method=self._hess_method).get_gradient(operator, params)
+
+        assert isinstance(params, tuple) and len(
```

Comment:
Can we replace this with a proper error-raising statement?

Reply:
Taking care of it.